temporalcli/client.go

```diff
@@ -255,7 +255,11 @@ func payloadCodecInterceptor(
 	)
 }
 
-func clientIdentity() string {
+// if providedIdentity is empty, use the hostname and username to generate a default identity
+func clientIdentity(providedIdentity string) string {
```

Comment:
if `--identity` is provided use that, otherwise use the custom version using hostname and username.

Reply:
I think we should consider making this a method on the `*ClientOptions` object. But really we should probably just get rid of this method, inline it in `dialClient` and document on `dialClient` that after the call is made successfully, `ClientOptions.Identity` is guaranteed to be non-empty and it should be used to access identity if needed. We saw the same thing with `ClientOptions.Namespace`.